This patch is based on the upstream commit described below, adapted for
use in the Debian package by Peter Michael Green.

commit 88a2a7a5a896d9b64d48b95f12e78cf91ee2b05f
Author: Paul Holzinger <pholzing@redhat.com>
Date:   Fri Feb 17 15:03:19 2023 +0100

    update netlink-packet-{route,core} to 0.15 and 0.5
    
    They contain breaking changes but nothing major, just moved same type
    definitions.
    
    Signed-off-by: Paul Holzinger <pholzing@redhat.com>

diff --git a/Cargo.toml b/Cargo.toml
index de0b465..2620f80 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -44,2 +44,2 @@ zbus = { version = "3.10.0" }
-netlink-packet-route = "0.13"
-netlink-packet-core = "0.4.2"
+netlink-packet-route = "0.15"
+netlink-packet-core = "0.5"
diff --git a/src/network/netlink.rs b/src/network/netlink.rs
index 4b11a28..dcf165a 100644
--- a/src/network/netlink.rs
+++ b/src/network/netlink.rs
@@ -9,11 +9,14 @@ use crate::{
     wrap,
 };
 use log::{info, trace};
+use netlink_packet_core::{
+    NetlinkHeader, NetlinkMessage, NetlinkPayload, NLM_F_ACK, NLM_F_CREATE, NLM_F_DUMP, NLM_F_EXCL,
+    NLM_F_REQUEST,
+};
 use netlink_packet_route::{
     nlas::link::{Info, InfoData, InfoKind, Nla},
-    AddressMessage, LinkMessage, NetlinkHeader, NetlinkMessage, NetlinkPayload, RouteMessage,
-    RtnlMessage, AF_INET, AF_INET6, IFF_UP, NLM_F_ACK, NLM_F_CREATE, NLM_F_DUMP, NLM_F_EXCL,
-    NLM_F_REQUEST, RTN_UNICAST, RTPROT_STATIC, RTPROT_UNSPEC, RT_SCOPE_UNIVERSE, RT_TABLE_MAIN,
+    AddressMessage, LinkMessage, RouteMessage, RtnlMessage, AF_INET, AF_INET6, IFF_UP, RTN_UNICAST,
+    RTPROT_STATIC, RTPROT_UNSPEC, RT_SCOPE_UNIVERSE, RT_TABLE_MAIN,
 };
 use netlink_sys::{protocols::NETLINK_ROUTE, SocketAddr};
 
@@ -369,10 +372,7 @@ impl Socket {
     }
 
     fn send(&mut self, msg: RtnlMessage, flags: u16) -> NetavarkResult<()> {
-        let mut packet = NetlinkMessage {
-            header: NetlinkHeader::default(),
-            payload: NetlinkPayload::from(msg),
-        };
+        let mut packet = NetlinkMessage::new(NetlinkHeader::default(), NetlinkPayload::from(msg));
         packet.header.flags = NLM_F_REQUEST | flags;
         packet.header.sequence_number = {
             self.sequence_number += 1;
@@ -440,6 +440,7 @@ impl Socket {
                             return Ok(result);
                         }
                     }
+                    _ => {}
                 };
 
                 offset += rx_packet.header.length as usize;
